code/modules/mob/living/simple_animal/abnormality/_abnormality.dm

```diff
@@ -125,9 +150,36 @@
 	else
 		gift_message += "\nYou are granted a gift by [src]!"
 
+	if(secret_chance && (prob(secret_chance)))
+		Initialize_secret_icon()
+
+/mob/living/simple_animal/hostile/abnormality/proc/Initialize_secret_icon()
+	SHOULD_CALL_PARENT(TRUE) // if you ever need to override this proc, consider adding onto it instead or not using all the variables given
+	secret_abnormality = TRUE
+
+	if(secret_icon_file)
+		icon = secret_icon_file
+
+	if(secret_icon_state)
+		icon_state = secret_icon_state
+
+	if(secret_icon_living)
+		icon_living = secret_icon_living
+
+	if(secret_horizontal_offset)
+		pixel_x = secret_horizontal_offset
+
+	if(secret_vertical_offset)
+		pixel_y = secret_vertical_offset
+
+	if(secret_total_offset)
+		pixel_x = secret_horizontal_offset
+		pixel_y = secret_vertical_offset
```

Comment:
Totally pointless given you have two vars for it already.

Reply:
tbh, thought we had more meme forms and it would actually be used. So fair